packages/core/src/external/carequality/ihe-gateway-v2/ihe-gateway-v2-async.ts

```diff
@@ -28,16 +31,26 @@ export class IHEGatewayV2Async extends IHEGatewayV2 {
     cxId: string;
   }): Promise<void> {
     const lambdaClient = makeLambdaClient(Config.getAWSRegion());
-    const params = { patientId, cxId, pdRequestGatewayV2 };
-    // intentionally not waiting
-    lambdaClient
-      .invoke({
-        FunctionName: iheGatewayV2OutboundPatientDiscoveryLambdaName,
-        InvocationType: "Event",
-        Payload: JSON.stringify(params),
-      })
-      .promise()
-      .catch(processAsyncError("Failed to invoke iheGatewayV2 lambda for patient discovery"));
+    const { gateways, ...rest } = pdRequestGatewayV2;
+
+    const chunks = Math.ceil(gateways.length / MAX_GATEWAYS_BEFORE_CHUNK);
+    const chunkSize = Math.ceil(gateways.length / chunks);
+    const gatewayChunks = chunk(gateways, chunkSize);
+
+    for (const chunk of gatewayChunks) {
+      const newPdRequestGatewayV2 = { ...rest, gateways: chunk };
+      const params = { ...newPdRequestGatewayV2, patientId, cxId };
```

Comment:
`params` contained a property called `pdRequestGatewayV2`, now we're deconstructing `newPdRequestGatewayV2`... won't it break?

Reply:
great catch. updated